Core/Core/LTI/LTITools.swift

```diff
@@ -89,27 +89,69 @@ public class LTITools: NSObject {
     var openInSafari: Bool { UserDefaults.standard.bool(forKey: "open_lti_safari") }
 
     public convenience init?(env: AppEnvironment = .shared, link: URL?) {
-        guard
-            let retrieve = link, retrieve.host == env.api.baseURL.host,
-            retrieve.path.hasSuffix("/external_tools/retrieve")
-        else { return nil }
+        guard let link, link.host == env.api.baseURL.host else { return nil }
 
-        let components = URLComponents.parse(retrieve)
+        if let (context, url, resourceLinkUUID) = Self.parseRegularExternalToolURL(url: link) {
+            self.init(
+                env: env,
+                context: context,
+                url: url,
+                resourceLinkLookupUUID: resourceLinkUUID
+            )
+        } else if let (courseID, toolID) = Self.parseQuerylessExternalToolURL(url: link) {
+            self.init(
+                env: env,
+                context: .course(courseID),
+                id: toolID,
+                launchType: .course_navigation
+            )
+            return
+        } else {
+            return nil
+        }
+
+    }
+
+    private static func parseRegularExternalToolURL(url: URL) -> (
+        context: Context,
+        url: URL?,
+        resourceLinkUUID: String?
+    )? {
+        guard url.path.hasSuffix("/external_tools/retrieve") else {
+            return nil
+        }
+        let components = URLComponents.parse(url)
 
-        let url: URL? = {
+        let newURL: URL? = {
             guard let urlQueryItem = components.queryValue(for: "url") else {
                 return nil
             }
             return URL(string: urlQueryItem)
         }()
         let resourceLinkUUID = components.queryValue(for: "resource_link_lookup_uuid")
 
-        if url == nil, resourceLinkUUID == nil {
+        if newURL == nil, resourceLinkUUID == nil {
             return nil
         }
 
-        let context = Context(url: retrieve) ?? .account("self")
-        self.init(env: env, context: context, url: url, resourceLinkLookupUUID: resourceLinkUUID)
+        let context = Context(url: url) ?? .account("self")
+        return (context, newURL, resourceLinkUUID)
+    }
+
+    // Parses LTI button triggered urls opened from K5 WebViews like Zoom and Microsoft LTIs.
+    // Expects a url without any query parameters: `courses/:courseID/external_tools/:toolID`.
+    // URLs with parameters are discarded, because there must be an already opened popup window loading the request.
+    private static func parseQuerylessExternalToolURL(url: URL) -> (courseID: String, toolID: String)? {
+        guard url.pathComponents.count == 5 else { return nil }
+
+        if let queryItems =  URLComponents.parse(url).queryItems, !queryItems.isEmpty {
+            return nil
+        } else {
```

Comment:
We could move these to the guard statement.

Reply:
We can't move it there because we generally don't want to any query items but if it's not nil we need to make sure that it's empty.

Reply:
Would this work in the guard? `(URLComponents.parse(url).queryItems ?? []).isEmpty`